extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java

```diff
@@ -1210,6 +1215,13 @@ private static void collectDialectConfig(String persistenceUnitName,
         if (dbProductVersion.isPresent()) {
             puPropertiesCollector.accept(AvailableSettings.JAKARTA_HBM2DDL_DB_VERSION, dbProductVersion.get());
         }
+
+        if ((dbKind.isPresent() && DatabaseKind.isPostgreSQL(dbKind.get())
+                || (dialect.isPresent() && dialect.get().toLowerCase(Locale.ROOT).contains("postgresql")))) {
+            reflectiveMethods.produce(new ReflectiveMethodBuildItem(
```

Comment:
This is really something that should be fixed upstream.

But in the meantime...

```suggestion
            // Workaround for https://hibernate.atlassian.net/browse/HHH-19063
            reflectiveMethods.produce(new ReflectiveMethodBuildItem(
```

Reply:
By the way, you might hit a similar issue with Oracle. I created https://hibernate.atlassian.net/browse/HHH-19064 upstream.